src/EnergyPlus/InputProcessing/CsvParser.cc

```diff
@@ -150,9 +119,13 @@ int CsvParser::find_number_columns(std::string_view csv, size_t &index)
         } else if (token == Token::DELIMITER) {
             ++num_columns;
         } else if (token == Token::LINE_END) {
-            ++num_columns;
+            // Catch a trailing comma, such as Shading files from E+ 22.2.0 and below
+            if (prev_token != Token::DELIMITER) {
+                ++num_columns;
+            }
```

Comment:
This is to keep supporting the workaround for Schedule:File:Shading from E+ 22.2.0 and below ( #9753)